spec/connectors/crawler/crawler_scheduler_spec.rb

```diff
@@ -156,6 +162,70 @@
         # it will return the base scheduling
         it_behaves_like 'triggers', nil
       end
+
+      context 'when base and custom scheduling are enabled and are scheduled after the poll interval' do
+        let(:sync_enabled) { true }
+        let(:weekly_enabled) { true }
+        let(:monthly_enabled) { true }
+
+        let(:next_trigger_time) { time_at_poll_start + poll_interval + 10 }
+        let(:weekly_next_trigger_time) { time_at_poll_start + poll_interval + 10 }
+        let(:monthly_next_trigger_time) { time_at_poll_start + poll_interval + 10 }
+
+        before(:each) do
+          allow(cron_parser).to receive(:next_time).with(time_at_poll_start).and_return(next_trigger_time, weekly_next_trigger_time, monthly_next_trigger_time)
+          expect(Utility::Logger).to receive(:debug).exactly(3).times.with(instance_of(String)) # expect three debug messages because three schedules are not being triggered
+        end
+
+        it_behaves_like 'does not trigger'
+      end
+
+      context 'when base and custom scheduling are enabled, but one is scheduled after the poll interval' do
+        let(:sync_enabled) { true }
+        let(:weekly_enabled) { true }
+
+        let(:next_trigger_time) { time_at_poll_start + poll_interval + 10 }
+        let(:weekly_next_trigger_time) { time_at_poll_start + poll_interval - 10 }
+
+        before(:each) do
+          allow(cron_parser).to receive(:next_time).with(time_at_poll_start).and_return(next_trigger_time, weekly_next_trigger_time)
+          expect(Utility::Logger).to receive(:debug).exactly(1).times.with(instance_of(String))
+        end
+
+        it_behaves_like 'triggers', :weekly_key
```

Comment:
Just an rspec point/nit: I don't think an `expect` should be in a `before` block if we're following principles of [four-phase testing](https://thoughtbot.com/blog/four-phase-test). The `before` block is part of the test's setup phase. We want to use `expect` during the verify stage, which is within the `it` block.

I can see how it looks awkward to add more work to the `it` block when this spec is using shared examples with `it_behaves_like`. It's possible to wrap the shared example in another `it` block. This will execute everything within as if it was one:

```suggestion
        before(:each) do
          allow(cron_parser).to receive(:next_time).with(time_at_poll_start).and_return(next_trigger_time, weekly_next_trigger_time)
        end

        it 'triggers the first schedule and skips the second' do
            expect(Utility::Logger).to receive(:debug).exactly(1).times.with(instance_of(String))
            it_behaves_like 'triggers', :weekly_key
        end

```

Reply:
So I'm running into a bit of a road block with this one - when I attempt to put an 'it_behaves_like' inside of an 'it' block, it gives me the following:

`RSpec::Core::ExampleGroup::WrongScopeError: `it_behaves_like` is not available from within an example (e.g. an `it` block) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc). It is only available on an example group (e.g. a `describe` or `context` block).`

I've tried the inverse of wrapping an `it`block with an `it_behaves_like do` block, but then the test doesn't behave as expected (expect won't receive a debug message)

Logically what you've suggested makes sense to me, but RSpec does not seem too happy. Thoughts?